src/main/java/org/opentripplanner/api/mapping/PlaceMapper.java

```diff
@@ -27,17 +28,22 @@ public static ApiPlace mapPlace(Place domain, Calendar arrival, Calendar departu
         ApiPlace api = new ApiPlace();
 
         api.name = domain.name;
-        api.stopId = FeedScopedIdMapper.mapToApi(domain.stopId);
-        api.stopCode = domain.stopCode;
-        api.platformCode = domain.platformCode;
+
+        if (domain.stop != null) {
+            api.stopId = FeedScopedIdMapper.mapToApi(domain.stop.getId());
+            api.stopCode = domain.stop != null ? domain.stop.getCode() : null;
+            api.platformCode = domain.stop instanceof Stop ? ((Stop) domain.stop).getPlatformCode() : null;
```

Comment:
What is the instanceOf check for? Can it be anything other than a `Stop`?

Reply:
Yes, It can be a FlexStopLocation if it is a result with a flex leg.